### DISCUSSION TITLE
Inverter definition - what does class get mapped to in Home Assistant

### DISCUSSION THREAD

**Discussion by gedger** [davidrapan/ha-solarman#69] [Q&A] [answered]
I've been going through improving my inverter definition which has raised a couple of questions.

What does "class:" get mapped to in HA? I had always presumed it was device_class but when I look at some of the other inverter definitions they use a class of switch which doesn't map to device_class based on the docs. In that case does it change the type from sensor to switch? But then a switch is an input device with a state not an output device.

https://www.home-assistant.io/integrations/binary_sensor/
https://www.home-assistant.io/integrations/sensor/

I wanted to used one of the binary sensor definitions, eg running, but the 0/1 result doesn't seem to map/work in HA as although the device_class is set to running the type is still a sensor rather than a binary_sensor.

Started me thinking that there should/needs to be device_class, device_type and state_class in the inverter definition with a direct mapping in HA to avoid any confusion?

**Comment by davidrapan** [comment, marked as answer]
Platform binary_sensor is not implemented.

R/W platforms (switch, select, time,..) uses class to distinguish their platform (except number which uses additional "configure:").

**Reply by gedger** [reply]
OK thanks, I couldn't find those answers so thanks. 

It's a confusing world, all of my inverter switches and are controlled by using write_coil and read_coil as they are single bits. Most of the single bit status flags are also only available by read_coil although a few are duplicated in a holding_registers. How far do you wish to take the integration to be compatible with more inverters like mine that required read-coil and some read only registers that must be read via read_input_register and then others which are readable and writable via read/write_holding_register. Also support of binary_sensors for those single bit on/off fields?

**Reply by davidrapan** [reply]
binary_sensor could be added I guess.

When comes to the R/W, reading is done as you are used to by defining that function code. Writing is for now done only by modbus function code 16. Extension to this functionality is done by per request basis, haha. 😆
